### PR TITLE
Improve toil cluster utils

### PR DESCRIPTION
**Pull request by natanlao**
* Utils will use aws provisioner by default

### CODE REVIEW

**Comment by cket** [comment]
@natanlao are you going to get both enhancements from the ticket in this PR?

**Comment by natanlao** [comment]
Jenkins, test this please

**Comment by cket** [comment]
@natanlao can you add a quick test to this? Just changing the provisioner utils test to use this default instead of explicitly setting `-p=aws` would be sufficient.

File: src/toil/test/provisioners/aws/awsProvisionerTest.py [hunk 1]
```diff
@@ -114,6 +114,10 @@ def _test(self, spotInstances=False):
                        '--provisioner=aws',
                        '--sseKey=/home/keyFile']
 
+        toilOptions.extend(['--nodeType=' + self.instanceType,
```
Comment (cket) [inline src/toil/test/provisioners/aws/awsProvisionerTest.py]: Why are you adding this in?

**Comment by cket** [comment]
Also, in the future can you do your work on a branch of this repo and not a fork? That way it's easier to collaborate on PRs since we will both have permissions for the repo.

File: src/toil/test/provisioners/aws/awsProvisionerTest.py [hunk 2]
```diff
@@ -96,8 +96,7 @@ def _test(self, spotInstances=False):
                        '--retryCount=2',
                        '--clusterStats=/home/']
 
-        toilOptions.extend(['--provisioner=aws',
```
Comment (cket) [inline src/toil/test/provisioners/aws/awsProvisionerTest.py]: This test will probably fail now, the default is only applicable to the cluster utilities - CGCloud is still an option for the actual autoscaling

**Comment by natanlao** [comment]
Jenkins, test this please

**Comment by cket** [comment]
@natanlao the build failures are due to hitting resource limits in AWS, you should probably hold off on new testing until I fix this

**Comment by cket** [comment]
jenkins, test this please

**Comment by cket** [comment]
jenkins, test this please

**Comment by cket** [comment]
jenkins, test this please

**Pull request merged**